### PR TITLE
src/fu-self-test: Make builder tests more resilient

### PR DESCRIPTION
**Pull request by superm1**
1) Skip firmware builder functions if missing CONFIG_USER_NS
2) Skip if detecting a patched kernel that has namespaces turned off
3) Don't fail if the error is -EPERM

Fixes FTBFS on Debian pbuilder and unprivileged docker.

### CODE REVIEW

File: src/fu-self-test.c [hunk 1]
```diff
@@ -1767,8 +1781,13 @@ fu_common_firmware_builder_func (void)
 						    "startup.sh",
 						    "firmware.bin",
 						    &error);
-	g_assert_no_error (error);
-	g_assert (firmware_blob != NULL);
+	if (firmware_blob == NULL) {
+		if (errno == EPERM) {
```
Comment (hughsie) [inline src/fu-self-test.c]: Can't we check the `@error` here rather than `EPERM`?

File: src/fu-self-test.c [hunk 2]
```diff
@@ -1754,6 +1755,19 @@ fu_common_firmware_builder_func (void)
 		g_test_skip ("no bwrap in path, so skipping");
 		return;
 	}
+	if (!g_file_test ("/proc/self/ns/user", G_FILE_TEST_IS_SYMLINK)) {
```
Comment (hughsie) [inline src/fu-self-test.c]: I kinda feel this needs to be split out into another function in this file -- perhaps we can even use it in the daemon?

File: src/fu-self-test.c (hunk 1 above)
Comment (superm1) [reply src/fu-self-test.c]: Fixed

File: src/fu-self-test.c (hunk 2 above)
Comment (superm1) [reply src/fu-self-test.c]: Split out and added to other places builder is called

**Comment by hughsie** [comment]
I've pushed a fixup commit; if that looks okay please squash it and merge. Thanks.

**Pull request merged**